src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Msmq/MsmqCommon.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Text;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Msmq
+{
+    internal class MsmqCommon
+    {
+        internal const string IntegrationName = nameof(IntegrationIds.Msmq);
+        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(IntegrationName);
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MsmqCommon));
+
+        internal static Scope CreateScope(Tracer tracer, string command, string spanKind, IMessageQueue messageQueue, bool? messagePartofTransaction = null)
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                var tags = new MsmqTags(spanKind)
+                {
+                    Command = command,
+                    IsTransactionalQueue = messageQueue.Transactional.ToString(),
```

Comment:
I don't know if the tags support `boolean` types directly but maybe we can avoid the string allocation here. cc. @kevingosse

Reply:
Yeah I added another comment wondering if we should store them as metrics instead. That said, I don't think `Boolean.ToString()` allocates, I need to check but I think the string is cached internally

Reply:
https://referencesource.microsoft.com/#mscorlib/system/boolean.cs,90

Reply:
Tags only support strings. Metrics support doubles, but they are not displayed in the UI. Anything we want to show to users as span metadata needs to be a string tag.

Reply:
thank you for your comments! so in the end is it ok to use bool.ToString, it doesnt seem like it's allocating memory but using constants right?

Reply:
Sounds about right. Thanks for the clarification Lucas.